client/gutenberg/extensions/simple-payments/edit.js

```diff
@@ -91,25 +91,20 @@ class SimplePaymentsEdit extends Component {
 		}
 
 		const { attributes, setAttributes } = this.props;
-		const { email, paymentId } = attributes;
-
-		this.setState( { isSavingProduct: true }, () => {
-			apiFetch( {
-				path: `/wp/v2/${ SIMPLE_PAYMENTS_PRODUCT_POST_TYPE }/${ paymentId ? paymentId : '' }`,
-				method: 'POST',
-				data: this.attributesToPost( attributes ),
-			} )
-				.then( response => {
-					const { id } = response;
-
-					if ( id ) {
-						setAttributes( { paymentId: id } );
-					}
+		const { email } = attributes;
+		const { saveEntityRecord } = dispatch( 'core' );
+
+		this.setState( { isSavingProduct: true }, async () => {
+			saveEntityRecord( 'postType', SIMPLE_PAYMENTS_PRODUCT_POST_TYPE, this.toApi() )
+				.then( record => {
```

Comment:
I was unable to find a way to use `withDispatch` for this need. Using `withDispatch`, I was unable to access the response from here in order to obtain the ID of the newly created post.